source/common/router/config_impl.cc

```diff
@@ -1924,10 +1954,6 @@ RouteConstSharedPtr VirtualHostImpl::getRouteFromEntries(const RouteCallback& cb
         if (match_status == RouteMatchStatus::Accept) {
           return route_entry;
         }
-        if (match_status == RouteMatchStatus::Continue &&
-            eval_status == RouteEvalStatus::NoMoreRoutes) {
-          return nullptr;
-        }
```

Comment:
This is a little refactoring. When match_status is Continue and eval_status is NoMoreRoutes, returning nullptr is the same as continuing the loop. After the refactoring, the code in matcher will be simpler.